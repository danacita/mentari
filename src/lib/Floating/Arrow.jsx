import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FloatingArrowWrapper, ArrowBounce } from './Styled';

const propTypes = {
  text: PropTypes.string,
  withArrow: PropTypes.bool
};

const defaultProps = {
  text: null,
  withArrow: null
};

class FloatingArrow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      scrolling: false
    };
    this.handleScroll = this.handleScroll.bind(this);
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  handleScroll() {
    const { scrolling } = this.state;
    if (window.scrollY === 0 && scrolling === true) {
      this.setState({ scrolling: false });
    } else if (window.scrollY > 180 && scrolling !== true) {
      this.setState({ scrolling: true });
    }
  }

  render() {
    const { text, withArrow } = this.props;
    const { scrolling } = this.state;
    return (
      <FloatingArrowWrapper hidden={scrolling} icon={withArrow}>
        <span>{text}</span>
        {withArrow ? (
          <ArrowBounce>
            <div className="arrow-down icon" />
          </ArrowBounce>
        ) : null}
      </FloatingArrowWrapper>
    );
  }
}

FloatingArrow.propTypes = propTypes;
FloatingArrow.defaultProps = defaultProps;

export default FloatingArrow;
