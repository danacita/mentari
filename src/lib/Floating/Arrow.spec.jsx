import React from 'react';
import renderer from 'react-test-renderer';

import Arrow from './Arrow';

describe('Arrow', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Arrow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
