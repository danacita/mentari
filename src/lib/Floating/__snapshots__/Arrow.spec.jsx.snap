// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow renders correctly 1`] = `
<div
  className="sc-bdVaJa fJIgxE"
  hidden={false}
>
  <span />
</div>
`;
